src/proxy/processors/push-action/clearBareClone.ts

```diff
@@ -1,16 +1,41 @@
 import { Action, Step } from '../../actions';
 import fs from 'node:fs';
 
+const WORK_DIR = './.remote/work';
+
 const exec = async (req: any, action: Action): Promise<Action> => {
   const step = new Step('clearBareClone');
 
-  // Recursively remove the contents of ./.remote and ignore exceptions
-  fs.rm('./.remote', { recursive: true, force: true }, (err) => {
-    if (err) {
-      throw err;
+  // In test environment, clean up EVERYTHING to prevent memory leaks
+  if (process.env.NODE_ENV === 'test') {
+    // TEST: Full cleanup (bare cache + all working copies)
+    try {
+      if (fs.existsSync('./.remote')) {
+        fs.rmSync('./.remote', { recursive: true, force: true });
+        step.log('Test environment: Full .remote directory cleaned');
+      } else {
+        step.log('Test environment: .remote directory already clean');
+      }
+    } catch (err) {
+      step.log(`Warning: Could not clean .remote directory: ${err}`);
+    }
+  } else {
```

Comment:
I feel like a safer option that ensures what we're testing more closely matches prod would be to either do some separate cleanup as necessary in the tests, or adjust config to have a much lower max (or even 0) then we can see if CacheManager itself is doing the cleanup.